### PR TITLE
Sprint 1: Tests

### PR DESCRIPTION
**Pull request by piowit**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -26,3 +26,11 @@ Lint/AmbiguousBlockAssociation:
...
 Style/Documentation:
   Enabled: false
+  
+RSpec/ExampleLength:
+  Exclude: 
+    - spec/system/*
+
+RSpec/MultipleExpectations:
+  Exclude: 
+    - spec/system/*
```
Comment (ciejj) [inline .rubocop.yml]: missing newline at end of file - I suggest configuring your IDE to take care of it for you ;)

File: spec/factories/kudos.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :kudo do
+    title { 'Title' }
+    content { 'Content' }
+    association :giver_of_kudo, factory: :employee
+    association :receiver_of_kudo, factory: :employee
```
Comment (ciejj) [inline spec/factories/kudos.rb]: [FYI] I think it is possible to write simply:
giver_of_kudo
receiver_of_kudo
but it is simply matter of preference - if you like it in explicit way it's fine ;)

File: spec/system/employee_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Employee account test', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  it 'create user, sign up, log out, log in' do
```
Comment (ciejj) [inline spec/system/employee_spec.rb:10]: I suggest splitting each action to separate test - sign up, and log in + log out
[edit] - after talking with Oskar I learned, that it is how he is teaching you, so please disregard this comment :)

File: spec/system/kudo_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+RSpec.describe 'Kudo test', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  let(:employee2) { build(:employee, email: 'test2@test.com') }
+  let(:employee3) { build(:employee, email: 'test3@test.com') }
+
+  it 'Setup 2 employees and crud kudo' do
```
Comment (ciejj) [inline spec/system/kudo_spec.rb:13]: Split those tests, so each kudo action is tested separately.

**Review by ciejj — CHANGES_REQUESTED** [review]
Good job, I left some suggestions- ping me on Slack if you get stuck with fixes ;)

File: spec/system/kudo_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    expect(page).to have_content 'Welcome! You have signed up successfully.'
+    click_link 'Log Out'
+    expect(page).to have_content 'Log in'
+
+    click_link 'Sign Up'
+    fill_in 'Email', with: employee3.email
+    fill_in 'Password', with: employee3.password
+    fill_in 'Password confirmation', with: employee3.password
+    click_button 'Sign up'
+    expect(page).to have_content 'Welcome! You have signed up successfully.'
```
Comment (ciejj) [inline spec/system/kudo_spec.rb]: why testing user singin/signnup in kudo tests? You already checked that in another spec :)

File: spec/system/kudo_spec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    click_link 'Edit'
+    fill_in 'Content', with: 'Another Content Test1'
+    click_button 'Update Kudo'
+    expect(page).to have_content 'Another Content Test1'
+
+    # accept_confirm do
+    click_link 'Destroy'
+    # end
+    expect(page).to have_content 'Kudo was successfully destroyed'
```
Comment (ciejj) [inline spec/system/kudo_spec.rb:34]: You are checking, notification, but what about page contents?

File: spec/system/kudo_spec.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    click_link 'New Kudo'
+    fill_in 'Title', with: 'Title test1'
+    fill_in 'Content', with: 'Content Test1'
+    click_button 'Create Kudo'
+
+    expect(page).to have_content 'Kudo was successfully created.'
+    expect(page).to have_content 'Content Test1'
+    expect(page).to have_content 'test3@test.com'
+
+    click_link 'Edit'
```
Comment (ciejj) [inline spec/system/kudo_spec.rb:40]: what if, there are mote than 1 kudo on the page?  How do you know, which one you are editing?

**Review by ciejj — CHANGES_REQUESTED** [review]

File: spec/system/kudo_spec.rb [hunk 8]
```diff
@@ -7,26 +7,15 @@
...
-    click_link 'Log Out'
-    expect(page).to have_content 'Log in'
-
-    click_link 'Sign Up'
-    fill_in 'Email', with: employee3.email
-    fill_in 'Password', with: employee3.password
-    fill_in 'Password confirmation', with: employee3.password
-    click_button 'Sign up'
-    expect(page).to have_content 'Welcome! You have signed up successfully.'
+    click_link 'Log In'
```
Comment (ciejj) [inline spec/system/kudo_spec.rb:14]: FYI - there is a helper, for that:
https://sajadtorkamani.com/how-to-login-in-capybara-tests-when-using-devise/

**Review by ciejj — APPROVED** [review]

File: .solargraph.yml [hunk 9]
```diff
@@ -0,0 +1,2 @@
+include:
+- "app/*.rb"y
```
Comment (oskarlakner) [inline .solargraph.yml]: this file shouldnt be included

**Comment by oskarlakner** [comment]
looks good 👍

**Comment by oskarlakner** [comment]
@piowit google `global git ignore`

File: .solargraph.yml (hunk 9 above)
Comment (piowit) [reply .solargraph.yml]: done

**Pull request merged**